src/Hash.h

```diff
@@ -13,34 +13,31 @@
 namespace TW::Hash {
 
 /// Enum selector for the supported hash functions
-enum HashFunc {
-    HashFunc_sha1 = 0, // SHA1
-    HashFunc_sha256, // SHA256
-    HashFunc_sha512, // SHA512
-    HashFunc_sha512_256, // SHA512/256
-    HashFunc_keccak256, // Keccak SHA256
-    HashFunc_keccak512, // Keccak SHA512
-    HashFunc_sha3_256, // version 3 SHA256
-    HashFunc_sha3_512, // version 3 SHA512
-    HashFunc_ripemd, // RIPEMD160
-    HashFunc_blake256, // Blake256
-    HashFunc_groestl512, // Groestl 512
-    HashFunc_sha256d, // SHA256 hash of the SHA256 hash
-    HashFunc_sha256ripemd, // ripemd hash of the SHA256 hash
-    HashFunc_sha3_256ripemd, // ripemd hash of the SHA256 hash
-    HashFunc_blake256d, // Blake256 hash of the Blake256 hash
-    HashFunc_blake256ripemd, // ripemd hash of the Blake256 hash
-    HashFunc_groestl512d, // Groestl512 hash of the Groestl512 hash
+enum Hasher {
+    Hash_sha1 = 0, // SHA1
```

Comment:
```suggestion
    HasherSha1 = 0, // SHA1
```

let's avoid `_` in enums